packages/user-mgnt/src/model/system.ts

```diff
@@ -26,15 +25,40 @@ export interface ISystem {
   status: string
   settings: {
     dailyQuota: number
+    webhook: WebHook[]
   }
   creationDate?: number
+  type: string
 }
 
+export enum EventType {
+  Birth = 'birth',
+  Death = 'death'
+}
+
+export interface WebHook {
+  event: EventType
+  permissions: WebhookPermissions[]
+}
+
+export enum WebhookPermissions {
+  CHILDS_DETAILS = 'childs-details',
+  MOTHERS_DETAILS = 'mothers-details',
+  FATHERS_DETAILS = 'fathers-details',
+  INFORMANT_DETAILS = 'informant-details',
+  DISEASE_DETAILS = 'disease-details'
+}
+
+const WebhookSchema = new Schema({
+  event: { type: String },
+  permissions: { type: [String] }
+})
+
 export interface ISystemModel extends ISystem, Document {}
 
 const systemSchema = new Schema({
-  name: { type: [UserNameSchema], required: true },
-  createdBy: { type: [UserNameSchema], required: true },
+  name: { type: String, required: true },
+  createdBy: { type: String, required: true },
```

Comment:
@euanmillar @Zangetsu101 

1) this `createdBy` could be a reference to User-model id: https://mongoosejs.com/docs/populate.html
2) Are Systems always before 1.2.0, and for now for upcoming versions, always made by the single natlsysadmin? I'm thinking about migrations. Can we just set the `createdBy` to be the single natlsysadmin for now?

If multiple users have been able to create systems earlier, I think there are destructive ways only to migrate this.

Reply:
The OpenCRVS documentation requires you to SSH in so you must be a system administrator. I reflected this here https://github.com/opencrvs/opencrvs-core/pull/4444/commits/c3b4c0f2f5089e32d14777978735b86fbcb41a77

Reply:
Before 1.2.0 it is always only the national system admin. I took a look at the PR you did for it and it looked good to me